client/web/src/repo/components/TryCodyWidget/TryCodyWidget.tsx

```diff
@@ -0,0 +1,182 @@
+                fill="#A305E1"
+            />
+        </g>
+        <defs>
+            <filter
+                id="filter0_dd_35_36617"
+                x="11.2822"
+                y="12.9186"
+                width="106.436"
+                height="102.163"
+                filterUnits="userSpaceOnUse"
+                colorInterpolationFilters="sRGB"
+            >
+                <feFlood floodOpacity="0" result="BackgroundImageFix" />
+                <feColorMatrix
+                    in="SourceAlpha"
+                    type="matrix"
+                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
+                    result="hardAlpha"
+                />
+                <feOffset dy="4" />
+                <feGaussianBlur stdDeviation="3" />
+                <feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.05 0" />
+                <feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow_35_36617" />
+                <feColorMatrix
+                    in="SourceAlpha"
+                    type="matrix"
+                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
+                    result="hardAlpha"
+                />
+                <feMorphology radius="3" operator="erode" in="SourceAlpha" result="effect2_dropShadow_35_36617" />
+                <feOffset dy="4" />
+                <feGaussianBlur stdDeviation="8.5" />
+                <feColorMatrix type="matrix" values="0 0 0 0 0.556863 0 0 0 0 0.207843 0 0 0 0 0.956863 0 0 0 0.42 0" />
+                <feBlend mode="normal" in2="effect1_dropShadow_35_36617" result="effect2_dropShadow_35_36617" />
+                <feBlend mode="normal" in="SourceGraphic" in2="effect2_dropShadow_35_36617" result="shape" />
+            </filter>
+        </defs>
+    </svg>
+)
+
+const AUTO_DISMISS_ON_EVENTS = new Set([
+    EventName.CODY_SIDEBAR_CHAT_OPENED,
+    EventName.CODY_SIDEBAR_EDIT,
+    EventName.CODY_SIDEBAR_RECIPE,
+    EventName.CODY_SIDEBAR_RECIPE_EXECUTED,
+    EventName.CODY_SIDEBAR_SUBMIT,
+])
+
+function useTryCodyWidget(telemetryService: TelemetryProps['telemetryService']): {
+    isDismissed: boolean | undefined
+    onDismiss: () => void
+} {
+    const [isDismissed = true, setIsDismissed] = useTemporarySetting('cody.blobPageCta.dismissed', false)
+
+    const onDismiss = useCallback(() => {
+        setIsDismissed(true)
+    }, [setIsDismissed])
+
+    // Listen for telemetry events to auto dismiss the widget
+    useEffect(() => {
+        if (isDismissed) {
+            return
+        }
+
+        return telemetryService.addEventLogListener?.(eventName => {
+            if (AUTO_DISMISS_ON_EVENTS.has(eventName as EventName)) {
+                onDismiss()
+            }
+        })
+    }, [telemetryService, isDismissed, onDismiss])
+
+    return { isDismissed, onDismiss }
+}
+
+export const TryCodyWidget: React.FC<TelemetryProps & { className?: string; type: 'blob' | 'repo' }> = ({
```

Comment:
To be consistent with other components where we define prop types in standalone interfaces.

```suggestion
interface TryCodyWidgetProps extends TelemetryProps {
  className?: string
  type: 'blob' | 'repo'
}

export const TryCodyWidget: React.FC<TryCodyWidgetProps> = ({
```